code/game/objects/items/devices/scanners.dm

```diff
@@ -541,18 +541,26 @@ REAGENT SCANNER
 	item_state = "analyzer"
 	origin_tech = "biotech=2"
 	w_class = WEIGHT_CLASS_SMALL
-	flags = CONDUCT
+	flags = CONDUCT | NOBLUDGEON
 	throwforce = 0
 	throw_speed = 3
 	throw_range = 7
 	materials = list(MAT_METAL=30, MAT_GLASS=20)
 
+/obj/item/slime_scanner/Initialize(mapload, ...)
+	. = ..()
+	new /datum/component/animation(src, list(COMSIG_ITEM_ATTACK = new /datum/animation_object(src, "adv_spectrometer_s_scan", 14)), \
```

Comment:
You'll want to use `AddComponent(...)` here I believe.

Reply:
Yes you're right. Will change!